checker/src/main/java/com/pushtorefresh/private_constructor_checker/PrivateConstructorChecker.java

```diff
@@ -138,16 +180,26 @@ private void check(Class clazz) {
 
         final Constructor<?> constructor = constructors[0];
 
-        if (constructor.getParameterTypes().length > 0) {
-            throw new AssertionError(clazz + " has non-default constructor with some parameters");
-        }
-
         constructor.setAccessible(true);
 
         if (!Modifier.isPrivate(constructor.getModifiers())) {
             throw new AssertionError("Constructor of " + clazz + " must be private");
         }
 
+        final Class<?>[] parameterTypes = constructor.getParameterTypes();
```

Comment:
So, it'll work only for classes with one private constructor? What is your use case of this? Just curious :smile: